src/hyp_map.rs

```diff
@@ -70,18 +72,122 @@ impl SharedRegion {
             let vaddr = PageAddr::new(RawAddr::supervisor_virt(r.base().bits())).unwrap();
             let page_count = PageSize::num_4k_pages(r.size()) as usize;
             let pte_fields = PteFieldBits::leaf_with_perms(pte_perms);
+            let variable_mapped = matches!(
+                r.region_type(),
+                HwMemRegionType::Reserved(HwReservedMemType::HypervisorHeap)
+                    | HwMemRegionType::Reserved(HwReservedMemType::HypervisorPerCpu)
+            );
```

Comment:
I think `variable_mapped` is meant to imply that the region can be mapped with huge-pages (see also Dylan's comment above). If that's the case, why would we restrict using hugepages to just these regions of memory? They're relatively quite small; the main benefit we get from using hugepages is mapping the rest of available DRAM or PCI MMIO space with hugepages.